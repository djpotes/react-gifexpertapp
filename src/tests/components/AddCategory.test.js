import AddCategory from "../../components/AddCategory"
import {shallow}  from "enzyme"

describe('Pruebas del component <AddCategory />', () =>{

    const setCategories = jest.fn();
    let wrapper;

    beforeEach( () => {
        wrapper = shallow(<AddCategory setCategories = {setCategories}/>);
    });
    test('Debe de mostrar el component correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Hola Mundo'
        input.simulate('change', { target: {value}});
        expect(wrapper.find('p').text().trim()).toBe(value);
    });

    test('No debe de postear la info obsubmit', () => {
        const form = wrapper.find('form');
        form.simulate('submit', {preventDefault(){}});
        expect(setCategories).not.toHaveBeenCalled();
    });

    test('Debe de llamar el setCategories y limpiar la caja de texto', () => {
        const value = 'Hola Mundo'
        wrapper.find('input').simulate('change', { target: {value}});
        wrapper.find('form').simulate('submit', {preventDefault(){}});
        //expect(setCategories).toHaveBeenCalledTimes(1);
        //expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
        expect(setCategories).toHaveBeenCalled();
        expect(wrapper.find('input').prop('value')).toBe('');
    });
})