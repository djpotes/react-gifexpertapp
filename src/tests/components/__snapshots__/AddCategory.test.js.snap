// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del component <AddCategory /> Debe de mostrar el component correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
